storage/src/main/java/tech/pegasys/teku/storage/server/rocksdb/dataaccess/V3RocksDbDao.java

```diff
@@ -113,6 +116,22 @@ public V3RocksDbDao(final RocksDbAccessor db) {
     return db.getAll(V3Schema.HOT_BLOCKS_BY_ROOT);
   }
 
+  @Override
+  public List<Bytes32> getStateRootsBeforeSlot(final UnsignedLong slot) {
+    try (Stream<ColumnEntry<Bytes32, SlotAndBlockRoot>> stream =
+        db.stream(V3Schema.STATE_ROOT_TO_SLOT_AND_BLOCK_ROOT)) {
+      return stream
+          .filter((column) -> column.getValue().getSlot().compareTo(slot) < 0)
+          .map(ColumnEntry::getKey)
+          .collect(Collectors.toList());
+    }
+  }
```

Comment:
If it becomes time consuming or difficult to support state roots on v3 databases we can skip it and only do it for v4/5.  Nothing critical depends on the lookup so it's reasonable for new features to only be available in the most recent database format.

Reply:
all good, currently its identical code...